api/v1alpha1/types.gen.go

```diff
@@ -85,6 +85,18 @@ const (
 	DeviceUpdatedStatusUpdating  DeviceUpdatedStatusType = "Updating"
 )
 
+// Defines values for GenericRepoSpecType.
+const (
+	GenericRepoSpecTypeGit  GenericRepoSpecType = "git"
+	GenericRepoSpecTypeHttp GenericRepoSpecType = "http"
+)
+
+// Defines values for HttpRepoSpecType.
+const (
```

Comment:
I'm not sure how the types get built into the `openapi.yaml`, but the current type definition results in some weird types:


```
import type { HttpConfig } from './HttpConfig';
export type HttpRepoSpec = {
  url: string;
  type: HttpRepoSpec.type;
  httpConfig: HttpConfig;
};
export namespace HttpRepoSpec {
  export enum type {
    GIT = 'git',
    HTTP = 'http',
  }
}
```
For each of the 3 repo spec types, we have a type, and also a namespace named equally. 
That means we have this name conflict, and that there are 3 different types defined for the "repoType".

Could we instead do something similar to the types for device statuses, etc?
See https://github.com/flightctl/flightctl/blob/main/api/v1alpha1/openapi.yaml#L2327 and https://github.com/flightctl/flightctl/blob/main/api/v1alpha1/openapi.yaml#L2332

IMO it would be better if we would generate a single "RepoSpecType" with values git / http, and the three repo spec types reference it.

Reply:
I like that approach. I'm not that familiar with the internals of openapi, but it makes sense to have it as a reference, and we avoid three different types of structs than need handling.
Thanks @celdrake !!

Reply:
It's perfect now :100: